deepchem/models/tensorgraph/layers.py

```diff
@@ -1398,6 +1400,363 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+  LSTM, and so allow step by step operation of the lstm
+  """
+
+  def __init__(self,
+               output_dim,
+               input_dim,
+               init='glorot_uniform',
+               inner_init='orthogonal',
+               activation='tanh',
+               inner_activation='hard_sigmoid',
+               **kwargs):
+
+    super(LSTMStep, self).__init__(**kwargs)
+
+    self.init = init
+    self.inner_init = inner_init
+    self.output_dim = output_dim
+
+    # No other forget biases supported right now.
+    self.activation = activation
+    self.inner_activation = inner_activation
+    self.input_dim = input_dim
+
+  def get_initial_states(self, input_shape):
+    return [model_ops.zeros(input_shape), model_ops.zeros(input_shape)]
+
+  def build(self):
+    init = initializations.get(self.init)
+    inner_init = initializations.get(self.inner_init)
+    self.W = init((self.input_dim, 4 * self.output_dim))
+    self.U = inner_init((self.output_dim, 4 * self.output_dim))
+
+    self.b = tf.Variable(
+        np.hstack((np.zeros(self.output_dim), np.ones(self.output_dim),
+                   np.zeros(self.output_dim), np.zeros(self.output_dim))),
+        dtype=tf.float32)
+    self.trainable_weights = [self.W, self.U, self.b]
+
+  def none_tensors(self):
+    W, U, b, out_tensor = self.W, self.U, self.b, self.out_tensor
+    h, c = self.h, self.c
+    trainable_weights = self.trainable_weights
+    self.W, self.U, self.b, self.out_tensor = None, None, None, None
+    self.h, self.c = None, None
+    self.trainable_weights = []
+    return W, U, b, h, c, out_tensor, trainable_weights
+
+  def set_tensors(self, tensor):
+    (self.W, self.U, self.b, self.h, self.c, self.out_tensor,
+     self.trainable_weights) = tensor
+
+  def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+    """Execute this layer on input tensors.
+
+    Parameters
+    ----------
+    in_layers: list
+      List of three tensors (x, h_tm1, c_tm1).
+
+    Returns
+    -------
+    list
+      Returns h, [h + c] 
+    """
+    activation = activations.get(self.activation)
+    inner_activation = activations.get(self.inner_activation)
+
+    self.build()
+    #if in_layers is None:
+    #  in_layers = self.in_layers
+    inputs = self._get_input_tensors(in_layers)
+    x, h_tm1, c_tm1 = inputs
+
+    # Taken from Keras code [citation needed]
+    z = model_ops.dot(x, self.W) + model_ops.dot(h_tm1, self.U) + self.b
+
+    z0 = z[:, :self.output_dim]
+    z1 = z[:, self.output_dim:2 * self.output_dim]
+    z2 = z[:, 2 * self.output_dim:3 * self.output_dim]
+    z3 = z[:, 3 * self.output_dim:]
+
+    i = inner_activation(z0)
+    f = inner_activation(z1)
+    c = f * c_tm1 + i * activation(z2)
+    o = inner_activation(z3)
+
+    h = o * activation(c)
+
+    if set_tensors:
+      self.h = h
+      self.c = c
+      self.out_tensor = h
+    return h, [h, c]
+
+
+def cos(x, y):
```

Comment:
Could we give this function a better name?  Any function that isn't computing a simple cosine should not be called `cos()`!  Also, is this meant to be a public function, or is it just for internal use?  If the latter, we should mark it as private with a `_`.